compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9996,24 +9996,49 @@ private STNode parseClientDeclOrVarDeclStatement(STNode metadata, List<STNode> q
     }
 
     private STNode parseClientDeclOrVarDecl(STNode metadata, STNode publicQualifier, List<STNode> qualifiers,
-                                            boolean isModuleVar) {
+                                            boolean moduleDecl) {
         STToken nextToken = peek();
         switch (nextToken.kind) {
             case STRING_LITERAL_TOKEN:
                 reportInvalidQualifiersOnClientDecl(publicQualifier, qualifiers);
-                return parseClientDeclaration(getAnnotations(metadata), qualifiers.get(qualifiers.size() - 1),
-                                              isModuleVar);
+                STNode clientKeyword = qualifiers.get(qualifiers.size() - 1);
+
+                if (metadata != null && metadata.kind == SyntaxKind.METADATA) {
+                    return parseModuleClientDeclaration((STMetadataNode) metadata, clientKeyword);
+                }
+
+                return parseClientDeclaration(getAnnotations(metadata), clientKeyword, moduleDecl);
             case OBJECT_KEYWORD:
-                if (isModuleVar) {
+                if (moduleDecl) {
                     return parseModuleVarDecl(metadata, publicQualifier, qualifiers);
                 }
 
                 return parseVariableDecl(getAnnotations(metadata), publicQualifier, new ArrayList<>(), qualifiers,
                                          false);
             default:
                 recover(nextToken, ParserRuleContext.CLIENT_DECL_OR_CLIENT_OBJECT_VAR_DECL);
-                return parseClientDeclOrVarDecl(metadata, publicQualifier, qualifiers, isModuleVar);
+                return parseClientDeclOrVarDecl(metadata, publicQualifier, qualifiers, moduleDecl);
+        }
+    }
+
+    private STNode parseModuleClientDeclaration(STMetadataNode metadata, STNode clientKeyword) {
+        STNode annotations = getAnnotations(metadata.annotations);
+        STNode documentationString = metadata.documentationString;
+
+        if (documentationString == null) {
+            return parseClientDeclaration(annotations, clientKeyword, true);
+        }
+
+        if (isNodeListEmpty(annotations)) {
+            clientKeyword =
+                    SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(clientKeyword, documentationString,
```

Comment:
```suggestion
            clientKeyword = SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(clientKeyword, documentationString,
```

Reply:
Exceeds length and indentation goes a bit off with the diagnostic code in the next line.

Reply:
Not for me :-/

Reply:
When aligning the arguments with the start of the argument in the method call in the previous line. Since the rest of the class seems to be following the other indentation will update this in the other PR.